### PR TITLE
Parsing

### PR DESCRIPTION
**Pull request by mmulich**
Parse the litezip file structure to python objects.

### CODE REVIEW

**Comment by mmulich** [comment]
[Parse litezip to objects](https://trello.com/c/1FOAL0ya/443-parse-litezip-to-objects)

File: litezip/main.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+Collection = namedtuple('Collection', 'id, file, resources')
+Module = namedtuple('Module', 'id, file, resources')
+
+
+COLLECTION_NSMAP = {
+    "bib": "http://bibtexml.sf.net/",
+    "c": "http://cnx.rice.edu/cnxml",
+    "col": "http://cnx.rice.edu/collxml",
+    "data": "http://www.w3.org/TR/html5/dom.html#custom-data-attribute",
+    "datadev": "http://dev.w3.org/html5/spec/#custom",
```
Comment (reedstrm) [inline litezip/main.py:30]: This made me think of it, but isn't the right place for the code. If we see anything w/ the datadev namespace, we should figure out how to update it to use the data namespace.

File: litezip/main.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+
+def _find_resources(directory, excludes=[]):
+    """Return a list of resource paths from the directory.
+    Ignore records via the list of `excludes`,
+    which are callables that take a file parameter (as a `Path` instance).
+
+    """
+    return [r for r in directory.glob('*')
+            if True not in [e(r) for e in excludes]]
```
Comment (reedstrm) [inline litezip/main.py:59]: cute!

File: litezip/main.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+    """
+    id = path.name
+    file = path / MODULE_FILENAME
+
+    if not file.exists():
+        raise MissingFile(file)
+
+    excludes = [
+        lambda filepath: filepath.name == MODULE_FILENAME,
```
Comment (reedstrm) [inline litezip/main.py:74]: There are a couple filenames we may want to consider excluding at this point - the generated index.cnxml.html and/or index.html.cnxml. I'll have to think if that's here or higher in the stack.

**Review by reedstrm — APPROVED** [review]

**Pull request merged**